tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -6,72 +6,76 @@ import (
-	session, err := NewSession(credentialPath, credAccount, &region)
+// New takes the generic type of information and returns an interface containing cloudProviderInfo specific to AWS.
+// credentialPath is a directory information to AWS credentials file.
+// credentialAccountID is the account name the user uses to create VM instance.
+// resourceTrackerDir is the path to where created instance and security group information is stored.
+func New(clientset *openshift.Client, credentialPath, credentialAccountID, resourceTrackerDir string) (cloudprovider.Cloud,
+	error) {
+	provider, err := clientset.GetCloudProvider()
 	if err != nil {
 		return nil, err
 	}
-	return &svc{ec2.New(session, aws.NewConfig()),
+	session, err := NewSession(credentialPath, credentialAccountID, provider.AWS.Region)
+	if err != nil {
+		return nil, err
+	}
+	return &cloudProviderInfo{ec2.New(session, aws.NewConfig()),
 		iam.New(session, aws.NewConfig()),
 		clientset,
-		&infraId,
-		dir,
+		resourceTrackerDir,
 	}, nil
 }
 
-// NewSession uses AWS credentials to create a session for interacting with AWS EC2
-// It returns an AWS session
-func NewSession(credPath, credAccount, region *string) (*sessionaws.Session, error) {
-	if _, err := os.Stat(*credPath); os.IsNotExist(err) {
-		return nil, fmt.Errorf("failed to find AWS credentials from path '%v'", *credPath)
+// NewSession uses AWS credentials to create and returns a session for interacting with EC2.
+func NewSession(credentialPath, credentialAccountID, region string) (*awssession.Session, error) {
+	if _, err := os.Stat(credentialPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("failed to find AWS credentials from path '%v'", credentialPath)
 	}
-	sess := sessionaws.Must(sessionaws.NewSession(&aws.Config{
-		Credentials: credentials.NewSharedCredentials(*credPath, *credAccount),
-		Region:      aws.String(*region),
+	session := awssession.Must(awssession.NewSession(&aws.Config{
+		Credentials: credentials.NewSharedCredentials(credentialPath, credentialAccountID),
+		Region:      aws.String(region),
 	}))
-	return sess, nil
+	return session, nil
 }
 
-// CreateWindowsVM creates an Windows VM instance based on a given AWS session under the same VPC as the existing OpenShift cluster
-// It attaches existing OpenShift cluster worker security group and IAM
-// It also uses public subnet, attaches public ip, and a security group that allows traffic from VPC range and RDP access from my IP
-// It uses given image id, instance type, and key name
-// It creates a unique name tag for the instance using the same prefix as the OpenShift cluster name
-// It logs id and security group information of the created instance
-// At the end, it provides RDP access information in the commandline interface
-func (s *svc) CreateWindowsVM(imageId, instanceType, keyName *string) error {
+// CreateWindowsVM creates an Windows VM instance under the same VPC as the existing OpenShift cluster,
```

Comment:
`a Windows instance`. If you want to list bullet points you cannot end with a `,`. You can saying something like, "with following properties:`. Note the `:` at the end.